# CompareEdge SQL Analysis

## Introduction
CompareEdge is a SaaS product comparison website that lists and compares
software tools across multiple categories. This analysis explores the
comparedge.com database to identify the most valuable product categories,
highest priced products, currency distribution and feature richness across
331 products and 5 interconnected tables.

## Dataset
- Source: comparedge_v1.sqlite (SQLite database)
- Tables: categories, products, pricing_plans, features, price_history
- Total categories: 29 rows
- Total Products: 331 rows
- Total Pricing Plans: 1,013 rows
- Total features: 6052 rows
- Total price_history: 916 rows

## Skills Used
- SQL (SQLite)
- Python (Pandas)
- Multi-table JOIN queries

In [5]:
#import libraries
import sqlite3
import pandas as pd

#connect to the database
conn = sqlite3.connect("comparedge_v1.sqlite")

#look what tables are inside
query = "SELECT name FROM sqlite_master WHERE type = 'table'"
tables = pd.read_sql(query, conn)
print(tables)

for table in ["categories", "products", "pricing_plans", "features", "price_history"]:
    query = f"SELECT COUNT(*) as rows FROM {table}"
    results = pd.read_sql(query, conn)
    print (f"{table}: {results['rows'][0]} rows")

#look whats inside each tables
for table in ["categories", "products", "pricing_plans", "features", "price_history"]:
    print(f"\n  {table}   ")
    query = f"SELECT * FROM {table} LIMIT 4"
    results = pd.read_sql(query, conn)
    print(results)

#Analysis
#Getting products names with their similar categories name.
query = "SELECT products.name, categories.name FROM products JOIN categories ON products.category_id = categories.id"
results = pd.read_sql(query, conn)
print(results)

#QN1.Which category with most products.
query = "SELECT categories.name, product_count FROM categories ORDER BY product_count DESC LIMIT 5"
results = pd.read_sql(query, conn)
print(results)

#QN2.Most expensive product monthly.
query = "SELECT products.name , price_monthly FROM pricing_plans JOIN products" \
" ON pricing_plans.product_id= products.id ORDER BY price_monthly DESC LIMIT 5"
results = pd.read_sql(query, conn)
print(results)

#QN3. Which currency appear mostly.
query = "SELECT currency,  COUNT(*) AS most_used FROM pricing_plans GROUP BY currency ORDER BY most_used DESC "
results = pd.read_sql(query, conn)
print(results)

#QN4.Which product has most features
query = """ SELECT products.name, COUNT(*) as features_counts FROM features JOIN products ON features.product_id
= products.id GROUP BY products.name ORDER BY features_counts DESC LIMIT 5"""
results = pd.read_sql(query, conn)
print(results)

#QN5.Category with highest average monthly price.
query = """SELECT categories.name, AVG(pricing_plans.price_monthly) AS AVRG_Price
FROM pricing_plans JOIN products ON pricing_plans.product_id = products.id
JOIN categories ON products.category_id = categories.id
GROUP BY categories.name ORDER BY AVRG_price DESC LIMIT 5 """
results = pd.read_sql(query, conn)
print(results)

#QN6.Product with biggest price range.
query = """SELECT products.name, MAX(price_history.price) - MIN(price_history.price) AS price_range
FROM price_history JOIN products ON price_history.product_id = products.id GROUP BY products.name ORDER BY price_range
 DESC LIMIT 5"""
results = pd.read_sql(query, conn)
print(results)

              name
0       categories
1         products
2    pricing_plans
3  sqlite_sequence
4         features
5    price_history
categories: 29 rows
products: 331 rows
pricing_plans: 1013 rows
features: 6052 rows
price_history: 916 rows

  categories   
   id             slug                   name  \
0   1         ai-tools               AI Tools   
1   2              llm  Large Language Models   
2   3    cloud-hosting          Cloud Hosting   
3   4  email-marketing        Email Marketing   

                                         description  product_count  
0  Complete directory of AI tools across writing,...            104  
1                                               None             25  
2                                               None             21  
3                                               None             20  

  products   
   id     slug        name  category_id  \
0   1   notion      Notion            5   
1   2  clickup     ClickUp            5   
2 

## Question 1 — Which Category Has The Most Products?

**Finding:**
AI Tools dominates with 104 products, way more than other categories,
followed by Large Language Models with 25 products. This indicates that
AI Tools is the biggest and most competitive category on comparedge.com.

## Question 2 — Which Products Are Most Expensive Monthly?

**Finding:**
HubSpot recorded the highest monthly price at \$3,600, followed by TokenTax at \$3,499 and Copy.ai at \$3,000. These premium pricedproducts represent
higher affiliate commission potential for comparedge.com and should be
prominently featured to maximize revenue per referral.

## Question 3 — Which Currency Appears Most in Pricing Plans?

**Finding:**
United States Dollar (USD) is the only currency used across all 1,013
pricing plans in the catalog. This indicates that comparedge.com's entire
product offering targets USD paying customers, suggesting the primary
audience is English speaking markets. Marketing and content efforts should
therefore focus on English speaking regions to maximize traffic and
affiliate conversions.

## Question 4 — Which Product Has The Most Features?

**Finding:**
Framer recorded the highest number of features with 51, followed by Miro
with 38, while vidIQ AI, v0 by Vercel and n8n tied at 20 features each.
Products with more features generate more detailed comparison pages,
attracting higher visitor engagement and increasing affiliate commission
opportunities for comparedge.com.

## Question 5 — Which Category Has The Highest Average Monthly Price?

**Finding:**
Crypto Tax recorded the highest average monthly price at $271.39, followed
by AI Writing at \$231.40 and Crypto Analytics at \$173.79. These premium
priced categories represent the highest affiliate commission potential per
referral, making them the most valuable categories to prioritize for content
creation and product featuring.

## Question 6 — Which Product Has The Biggest Price Range?

**Finding:**
HubSpot recorded the largest price range at \$3,600, followed by TokenTax
at \$3,450 and Shopify at \$2,262. Products with wider price ranges appeal
to a broader audience — from free users to enterprise clients — generating
more comparison traffic on comparedge.com and increasing the likelihood of
affiliate conversions across multiple customer segments.

## Summary

This analysis of the comparedge.com database reveals three critical insights.
First, AI Tools is the dominant category with 104 products, significantly
ahead of all others, suggesting the website should prioritize featuring and
expanding AI tool listings to capitalize on this demand. Second, USD is the
only currency across all pricing plans, indicating that marketing and content
efforts should focus on English speaking and USD paying markets for maximum
affiliate conversion. Finally, products like HubSpot and TokenTax with the
highest monthly prices and widest price ranges represent the greatest affiliate
commission potential and should be prominently featured across the website.